components/integrations/confirm_integration/confirm_integration.jsx

```diff
@@ -58,6 +59,7 @@ export default class ConfirmIntegration extends React.Component {
         const outgoingHook = this.props.outgoingHooks[this.state.id];
         const oauthApp = this.props.oauthApps[this.state.id];
         const bot = this.props.bots[this.state.id];
+        const siteURL = this.props.siteURL || getSiteURL();
```

Comment:
Maybe `getSiteURL()` is set in the `mapStateToProps` instead?

Reply:
Sure thing. You mean, as a function prop, or just move the `... || getSiteURL` to `mapStateProps`?

Reply:
Yeah, my proposal would be to move the `... || getSiteURL` to `mapStateToProps`, increasing the purity of the component as a measure of whether its output is fully defined by its input props.